src/sonic-bgpcfgd/bgpcfgd/managers_device_global.py

```diff
@@ -142,6 +150,54 @@ def configure_idf(self, data=None):
         else:
             log_notice("DeviceGlobalCfgMgr:: IDF configuration is up-to-date")
 
+    def configure_received_bandwidth(self, data=None):
+        """Configure received bandwidth for W-ECMP feature"""
+        
+        state = self.RECEIVED_DEFAULTS
+        
+        if data is not None:
+            if "received_bandwidth" in data:
+                state = data["received_bandwidth"]
+        
+        if self.is_update_required("received_bandwidth", state):
+            if self.set_received_bandwidth(state):
+                self.directory.put(self.db_name, self.table_name, "received_bandwidth", state)
+        else:
+            log_notice("DeviceGlobalCfgMgr:: received bandwidth for W-ECMP configuration is up-to-date")
+     
+    def set_received_bandwidth(self, status):
+        """API to configure received bandwidth for W-ECMP"""
+        
+        bgp_asn = self.directory.get_slot("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME)["localhost"]["bgp_asn"]
+        cmd_list = []
+        
+        if status not in ["ignore", "allow", "skip_missing", "default_weight_for_missing"]:
+            log_err("Received Bandwidth for W-ECMP: invalid value({}) is provided".format(status))
+            return False
+        
+        cmd_list.append("router bgp %s" % bgp_asn)
+        
+        if status == "ignore":
+            log_notice("DeviceGlobalCfgMgr:: Ignoring received_bandwdith...")
```

Comment:
@colintle please add W-ECMP label to all relevant messages

Reply:
Done